File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/Go/s012761786_commented.go
package main

import (
	"fmt"
	"math"
	"bufio"
	"os"
	"strconv"
	"strings"
)

var sc = bufio.NewScanner(os.Stdin)

func main() {
	sc.Split(bufio.ScanWords)

	n := nextLong()
	a := nextLong()
	b := nextLong()

	sa := int64(math.Abs(float64(a - b)))

	ans := int64(0)

	if sa%2 == 0 {
		ans = sa / 2
	} else {
		AAA := int64(math.Min(float64(n-a), float64(a-1)))
		BBB := int64(math.Min(float64(n-b), float64(b-1)))

		ans = int64(math.Min(float64(AAA), float64(BBB))) //片方が端っこに来るまで。

		//差を1減らす⇒差が偶数になる
		sa -= 1
		ans += 1

		//差は偶数
		ans += sa / 2
	}

	fmt.Println(ans)
}

func next() string {
	sc.Scan()
	return sc.Text()
}

func nextInt() int {
	sc.Scan()
	i, e := strconv.Atoi(sc.Text())
	if e != nil {
		panic(e)
	}
	return i
}

func nextLong() int64 {
	sc.Scan()
	i, e := strconv.ParseInt(sc.Text(), 10, 64)
	if e != nil {
		panic(e)
	}
	return i
}

func nextFloat() float64 {
	sc.Scan()
	i, e := strconv.ParseFloat(sc.Text(), 64)
	if e != nil {
		panic(e)
	}
	return i
}

func nextString() string {
	sc.Scan()
	return sc.Text()
}

func splitString(s string) []string {
	return strings.Split(s, "")
}

func splitChar(s string) []rune {
	return []rune(s)
}

func charToInt(c byte) int {
	return int(c - '0')
}

func maxInt(a []int, lng int) int {
	max := a[0]
	for i := 1; i < lng; i++ {
		if a[i] > max {
			max = a[i]
		}
	}
	return max
}

func minInt(a []int, lng int) int {
	min := a[0]
	for i := 1; i < lng; i++ {
		if a[i] < min {
			min = a[i]
		}
	}
	return min
}

func maxLong(a []int64, lng int) int64 {
	max := a[0]
	for i := 1; i < lng; i++ {
		if a[i] > max {
			max = a[i]
		}
	}
	return max
}

func minLong(a []int64, lng int) int64 {
	min := a[0]
	for i := 1; i < lng; i++ {
		if a[i] < min {
			min = a[i]
		}
	}
	return min
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/uncomment/direct/Java/C++/s012761786_commented.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

string next();
int nextInt();
long long nextLong();
double nextDouble();
vector<string> splitString(const string&);
vector<char> splitChar(const string&);
int charToInt(char);
int maxInt(int[], int);
int minInt(int[], int);
long long maxLong(long long[], int);
long long minLong(long long[], int);

int main() {
    long long n = nextLong();
    long long a = nextLong();
    long long b = nextLong();

    long long sa = std::llabs(a - b);

    long long ans = 0;

    if (sa % 2 == 0) {
        ans = sa / 2;
    }
    else {
        long long AAA = std::min((n - a), (a - 1LL));
        long long BBB = std::min((n - b), (b - 1LL));

        ans = std::min(AAA, BBB);

        sa -= 1;
        ans += 1;

        ans += sa / 2;
    }

    cout << ans << endl;

    return 0;
}

string next() {
    string s;
    cin >> s;
    return s;
}

int nextInt() {
    int val;
    cin >> val;
    return val;
}

long long nextLong() {
    long long val;
    cin >> val;
    return val;
}

double nextDouble() {
    double val;
    cin >> val;
    return val;
}

vector<string> splitString(const string& s) {
    vector<string> result;
    for (char c : s) {
        result.push_back(string(1, c));
    }
    return result;
}

vector<char> splitChar(const string& s) {
    return vector<char>(s.begin(), s.end());
}

int charToInt(char c) {
    return c - '0';
}

int maxInt(int arr[], int lng) {
    int max = arr[0];
    for (int i = 1; i < lng; i++) {
        max = std::max(max, arr[i]);
    }
    return max;
}

int minInt(int arr[], int lng) {
    int min = arr[0];
    for (int i = 1; i < lng; i++) {
        min = std::min(min, arr[i]);
    }
    return min;
}

long long maxLong(long long arr[], int lng) {
    long long max = arr[0];
    for (int i = 1; i < lng; i++) {
        max = std::max(max, arr[i]);
    }
    return max;
}

long long minLong(long long arr[], int lng) {
    long long min = arr[0];
    for (int i = 1; i < lng; i++) {
        min = std::min(min, arr[i]);
    }
    return min;
}

// End of Code
